torch/distributed/pipelining/PipelineStage.py

```diff
@@ -575,12 +578,8 @@ def backward_one_chunk(
                 "input_values": input_values,
             }
 
-        self.grads_input = self.backward_maybe_with_nosync(
-            bwd_kwargs, self.bwd_chunk_id
-        )
-        logger.debug(
-            f"{self.log_prefix} Backwarded chunk {self.bwd_chunk_id}"  # noqa: G004
-        )
+        self.grads_input = self.backward_maybe_with_nosync(bwd_kwargs, bwd_chunk_id)
+        logger.debug(f"{self.log_prefix} Backwarded chunk {bwd_chunk_id}")  # noqa: G004
         self.bwd_chunk_id += 1
```

Comment:
its a little weird that we would keep incrementing self.bwd_chunk_id inside the stage but also expose an override kwarg that ignores the internal id.

this is a bit of an API issue with stage. maybe we should rename to `backward_chunk()` and require the chunk ID as part of the api name?  We could leave forward_one_chunk alone if we think users always want to go in order for forward.  @kwen2501 wdyt?

Reply:
yeah adding the optional arg was sort of a bandaid so that I didn't need to make updates in other places as part of the PR. It's dangerous if someone accidentally mixes it up if they are calling `backward_one_chunk` with `chunk_id` and also calling it without it in the same function.

Open to suggestions, if we want to keep the naming the same we could also just make the arg a requirement

Reply:
Ill land this after https://github.com/pytorch/pytorch/pull/127935, which will resolve the issue